pkg/router/controller/secret_manager.go

```diff
@@ -0,0 +1,68 @@
+package controller
+
+import (
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/watch"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"github.com/openshift/library-go/pkg/route/secret"
+	"github.com/openshift/router/pkg/router"
+)
+
+type SecretManagerLoader struct {
+	// plugin is the next plugin in the chain.
+	plugin router.Plugin
+
+	secretManager *secret.Manager
+
+	// recorder is an interface for indicating route rejections.
+	recorder RejectionRecorder
+}
+
+func NewSecretManagerLoader(plugin router.Plugin, secretmanager *secret.Manager, recorder RejectionRecorder) *SecretManagerLoader {
+	return &SecretManagerLoader{
+		plugin:        plugin,
+		secretManager: secretmanager,
+		recorder:      recorder,
+	}
+}
+
+func (sm *SecretManagerLoader) HandleRoute(eventType watch.EventType, route *routev1.Route) error {
```

Comment:
What currently handles updating routes when the service cert changes.  That piece of logic needs to do a 

```go
func (c *controller) queueRoute(route, oldRoute ){
  if route.RemovedOldExternalSecret(){
     secretmonitor.RemoveEventHandler(savedTheEventMonitorByRoute)
  }
  if route.hasNewExternalSecret(){
    saveTheEventMonitorByRoute := secretmonitor.AddEventHanlder(cache.ResourceEventHandlerFuncs{
      AddFunc:    func(obj interface{}) {
        c.queue.Add(route)
      },
      UpdateFunc: func(oldObj, newObj interface{}) {
        c.queue.Add(route)
      },
      DeleteFunc: func(obj interface{}) {
        c.queue.Add(route)
      },
    })
}
```